python-package/lightgbm/basic.py

```diff
@@ -195,6 +187,7 @@ def __init__(self, model_file=None, booster_handle=None, is_manage_handle=True):
         """
         self.handle = ctypes.c_void_p()
         self.__is_manage_handle = True
+        self.init_iteration = init_iteration
```

Comment:
Actually, init_iteration will be reset by out_num_iterations. 
So not need to pass this parameter.
you can just let self.__num_total_iteration to public.